Add Microchip mec_assy6941 evaluation board for MEC5 HAL based MEC174x and MEC175x chips

scottwcpg opened the pull request:
Ran check_compliance.py and twister and all passed.

scottwcpg committed:
dts: arm: microchip: Fixes in preparation for boards

We fixed some minor bugs and one missing component for
Microchip MEC5 HAL based chips, MEC174x and MEC175x.
These changes are in preparation for board check-in and
hello world sample. Added the clock-frequency property
to the cpu and rtimer nodes. This properity is used
derive the Kconfig's for the kernel tick.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

scottwcpg committed:
boards: microchip: mec_assy6941: Add MEC5 evaluation board 6941

We added the mec_assy6941 an evaluation board for MEC5 HAL based
MEC174x and MEC175x parts. We support these Microchip chips:
mec1743_qlj, mec1743_qsz, mec1753_qlj, and mec1753_qsz. LJ refers
to 176-pin and SZ to 144-pin WFGA packages. Q refers to total SRAM
of 480KB. Building the board with one of the SoC's requires passing
mec_assy9641/<soc_name>. For example: mec_assy6941/mec1753_qlj is
passed to west via the "-b" command line option.
The SoC's are configured to use Microchip's 32-bit 32KHz RTOS timer
as the kernel tick. Refer to the board DTS rtimer node. If the user
wishes to use ARM SYSTICK as the kernel timer then disable the rtimer
node in the application overlay.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

scottwcpg committed:
drivers: serial: Microchip MEC5 add select PINCTRL to Kconfig

Zephyr has changed the rules for PINCTRL to use Kconfig select
feature at the driver level instead of setting the Kconfig
item directly in the board or application level. We updated
the MEC5 UART serial driver Kconfig to select PINCTRL.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

nordicjm commented on boards/microchip/mec_assy6941/Kconfig.defconfig:
```diff
@@ -0,0 +1,28 @@
+
+if RTOS_TIMER
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 32768
```
should come from dts property

nordicjm commented on boards/microchip/mec_assy6941/Kconfig.defconfig:
```diff
@@ -0,0 +1,28 @@
+	BOARD_MEC_ASSY6941_MEC1753QLJ || BOARD_MEC_ASSY6941_MEC1753QSZ
+
+if RTOS_TIMER
+config SYS_CLOCK_HW_CYCLES_PER_SEC
```
newline gap after if

scottwcpg replied:
Removed.  Kernel timer configs moved to soc layer.

nordicjm replied:
the changes need squashing into the correct commit, the original commit still has this and is wrong

scottwcpg replied:
Squashed mec_assy6941 commits into one. Should be resolved.

nordicjm commented on boards/microchip/mec_assy6941/Kconfig.defconfig:
```diff
@@ -0,0 +1,28 @@
+
+endif # RTOS_TIMER
+
+if !RTOS_TIMER
```
as above

scottwcpg replied:
Removed.  Kernel timer configs moved to soc layer.

nordicjm commented on boards/microchip/mec_assy6941/Kconfig.defconfig:
```diff
@@ -0,0 +1,28 @@
+# Refer to SOC_MEC_PROC_CLK_DIV
+#
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 96000000
```
as above, if one of these is dts and the other is not then ok but at least one (or both) should be dts entries

scottwcpg replied:
Moved to soc layer.  Value based on MCHP timer driver DT enable.

nordicjm commented on boards/microchip/mec_assy6941/Kconfig.defconfig:
```diff
@@ -0,0 +1,28 @@
+	default 1000
+
+endif # !RTOS_TIMER
+endif # BOARD_MEC_ASSY6941_MEC1743QLJ or _MEC1743QSZ or _MEC1753QLJ or _MEC1753QSZ
```
newline gap after and use full Kconfig names

scottwcpg replied:
Moved to soc layer.

nordicjm commented on boards/microchip/mec_assy6941/board.yml:
```diff
@@ -0,0 +1,17 @@
+boards:
+  - name: mec_assy6941_mec1743qlj
+    vendor: microchip
+    socs:
+      - name: mec1743_qlj
+  - name: mec_assy6941_mec1743qsz
+    vendor: microchip
+    socs:
+      - name: mec1743_qsz
+  - name: mec_assy6941_mec1753qlj
+    vendor: microchip
+    socs:
+      - name: mec1753_qlj
+  - name: mec_assy6941_mec1753qsz
+    vendor: microchip
+    socs:
+      - name: mec1753_qsz
```
whilst you can do this, I don't see why you wouldn't just add 1 board with 4 socs then build for e.g. `mec_assy6941/mec1743_qsz`

scottwcpg replied:
Thanks for the hint.  Changed it to be one board with choice of 1 of 4 socs.

nordicjm commented on boards/microchip/mec_assy6941/doc/index.rst:
```diff
@@ -0,0 +1,434 @@
+Supported Features
+==================
+
+The mec5_assy6941 board configuration supports the following hardware
```
use new hw directive tag

scottwcpg replied:
Added HW directive tag.

nordicjm commented on boards/microchip/mec_assy6941/doc/mec5_assy6941_evb.jpg:
convert to webp then put through https://tinypng.com/

scottwcpg replied:
Done.  Only a slight reduction in file sizes.

scottwcpg committed:
soc: microchip: mec: Unify kernel timer tick rate configuration

Microchip MEC was configuring the kernel timer tick rate at the
board level instead of the SoC level. We unify all this by moving
the Kconfig logic out of the board level into each mec chip.
We also derive SYS_CLOCK_HW_TICKS_PER_SEC from the device tree
node enabled for the kernel timer: Microchip's 32 KHz RTOT timer
or Cortex-M4 SYSTICK. The soc kconfig rules are loading all mec
subdirectories for every build causing warnings when building
new socs with this change. We made the changes for all the mec chips.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

scottwcpg committed:
boards: microchip: mec1501modular_assy6885: Remove kernel timer logic

Microchip MEC kernel timer configuration logic has been moved to the
soc layer.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

scottwcpg committed:
boards: microchip: mec15xxevb_assy6885: Remove kernel timer kconfig logic

Kconfig logic for kernel timer configuration and selection has been
moved to the soc layer.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

scottwcpg committed:
boards: microchip: mec172xevb_assy6906: Kernel timer config moved to soc

We moved kernel timer kconfig logic to the soc layer.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

scottwcpg committed:
boards: microchip: mec172xmodular_assy6930: Kernel timer logic moved to soc

We moved kernel timer configuration logic to the soc layer.
In addition, removed zephyr library cmake rule causing a build
warning since there are no C source files in the board.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

scottwcpg commented:
NOTE: The changes moving kernel timer selection to soc layer caused build warning for the older mec15xx and mec17xx non-HAL based chips. I expanded this PR to 10 commits to fix these warnings.

scottwcpg commented on boards/microchip/mec_assy6941/doc/index.rst:
```diff
@@ -0,0 +1,407 @@
+:zephyr_file:`boards/microchip/mec_assy6941/mec_assy6941_mec1743_qlj_defconfig` Kconfig file.
+:zephyr_file:`boards/microchip/mec_assy6941/mec_assy6941_mec1743_qsz_defconfig` Kconfig file.
+:zephyr_file:`boards/microchip/mec_assy6941/mec_assy6941_mec1753_qlj_defconfig` Kconfig file.
+:zephyr_file:`boards/microchip/mec_assy6941/mec_assy6941_mec1753_qsz_defconfig` Kconfig file.
```
Thanks. Updated index.rst removing recommended lines and checked it with doc html build.

scottwcpg commented on boards/microchip/mec_assy6941/doc/index.rst:
```diff
@@ -0,0 +1,407 @@
+.. image:: mec_assy6941_evb.webp
+     :align: center
+     :alt: MEC5_ASSY6941 EVB
+
```
Thanks. Updated index.rst removing recommended lines and checked it with doc html build.

nordicjm commented on boards/microchip/mec_assy6941/CMakeLists.txt:
```diff
@@ -0,0 +1,60 @@
+  set(MEC5_SPI_GEN $ENV{MEC5_SPI_GEN})
+else()
+  # Else find the tool in PATH
+
```
```suggestion
```

scottwcpg replied:
Done

nordicjm commented on boards/microchip/mec_assy6941/Kconfig:
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2025 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_MEC_ASSY6941
```
empty file to go

scottwcpg replied:
Done

nordicjm commented on soc/microchip/mec/mec15xx/Kconfig.defconfig.series:
```diff
@@ -13,17 +13,25 @@ config NUM_IRQS
-if RTOS_TIMER
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 32768 if MCHP_XEC_RTOS_TIMER
+	default 96000000
```
this should be a dts property which is fetched

scottwcpg commented:
@albertofloyd Hi Alberto, this is a PR for the MEC175x and MEC175x EVB boards.  Could you review? Thanks.

kartben commented on boards/microchip/mec_assy6941/board.yml:
```diff
@@ -0,0 +1,9 @@
+board:
+  name: mec_assy6941
+  full_name: MEC174x/5x EVB
```
To follow the naming of other boards I think it might be better to do as follows?

```suggestion
  full_name: MEC17xxEVB ASSY6941
```

scottwcpg replied:
Done. Pushed update.

kartben reviewed (approved):
+1 for docs

kartben commented:
@nordicjm please refresh your +1 - thanks!

scottwcpg commented:
@kartben Hi, thanks for the pointer.  Jay is my co-working in the same office.  I will talk to him.
